libraries/Microsoft.Bot.Connector/Authentication/MicrosoftGovernmentAppCredentials.cs

```diff
@@ -62,5 +65,34 @@ public override string OAuthEndpoint
         {
             get { return GovernmentAuthenticationConstants.ToChannelFromBotLoginUrl; }
         }
+
+        protected override Lazy<IAuthenticator> BuildIAuthenticator()
```

Comment:
I cannot follow this method, and all of it's nested lambda functions. 

Please simplify.